code/__HELPERS/announcements.dm

```diff
@@ -0,0 +1,78 @@
+/**
+ * Sends a div formatted chat box announcement
+ *
+ * Formatted like:
+ *
+ * " Server Announcement " (or sender_override)
+ *
+ * " Title "
+ *
+ * " Text "
+ *
+ * Arguments
+ * * text - required, the text to announce
+ * * title - optional, the title of the announcement.
+ * * players - optional, a list of all players to send the message to. defaults to the entire world
+ * * play_sound - if TRUE, play a sound with the announcement (based on player option)
+ * * sound_override - optional, override the default announcement sound
+ * * sender_override - optional, modifies the sender of the announcement
+ * * encode_title - if TRUE, the title will be HTML encoded
+ * * encode_text - if TRUE, the text will be HTML encoded
+ * * color_override - optional, set a color for the announcement box
+ */
+
+/proc/send_formatted_announcement(text, title = "", players, play_sound = TRUE, sound_override, sender_override, encode_title = TRUE, encode_text = TRUE, color_override)
+	if(isnull(text))
+		return
+
+	var/list/announcement_strings = list()
+
+	if(encode_title && title && length(title) > 0)
+		title = html_encode(title)
+		if(encode_text)
+			text = html_encode(text)
+			if(!length(text))
+				return
+
+		if(isnull(sender_override))
+			sender_override = "Server Admin Announcement"
+
+		if(isnull(sound_override)) // this is not an IC announcement, so we don't adjust for announcer station traits
+			sound_override = 'sound/ai/default/attention.ogg'
+
+		if(isnull(color_override))
+			color_override = "grey"
```

Comment:
It'll make for a long signature but I don't see why these couldn't just be default values for these arguments, like the booleans already have